executor/write.go

```diff
@@ -1225,8 +1225,12 @@ func (e *InsertExec) batchUpdateDupRows(newRows [][]types.Datum, onDuplicate []*
 
 // fillBackKeys fills the updated key-value pair to the dupKeyValues for further check.
 func (e *InsertExec) fillBackKeys(fillBackKeysInRow []keyWithDupError, handle int64) {
-	e.dupOldRowValues[string(e.Table.RecordKey(handle))] = fillBackKeysInRow[0].newRowValue
+	filled := false
```

Comment:
How about we check the length of `fillBackKeysInRows` and return if zero?

Reply:
PTAL @coocood